locker/src/main/java/org/killbill/commons/locker/zookeeper/recipes/ProtocolSupport.java

```diff
@@ -0,0 +1,186 @@
+/*
```

Comment:
For all these recipes files:
1. In there a small jar we could import instead of copying the classes -- if there is none, maybe this is ok.
2. If we go with copying classes, we should **keep the original copyright** . Also we should document the version so we can later update such files if needed.